lockfile/src/cargo.rs

```diff
@@ -0,0 +1,164 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+use serde::Deserialize;
+
+use crate::{Parse, ParseResult};
+
+pub struct Cargo;
+
+#[derive(Deserialize, Debug, Clone)]
+struct CargoLock {
+    #[serde(rename = "package")]
+    packages: Vec<Package>,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+struct Package {
+    name: String,
+    version: String,
+    source: Option<String>,
+}
+
+impl Parse for Cargo {
+    /// Parses `cargo.lock` files into a vec of packages
+    fn parse(&self, data: &str) -> ParseResult {
+        let mut lock: CargoLock = toml::from_str(data)?;
+        Ok(lock
+            .packages
+            .drain(..)
+            .filter_map(|package| {
+                package.source.as_ref().map(|_| PackageDescriptor::from(package.clone()))
+            })
+            .collect())
+    }
+
+    fn package_type(&self) -> PackageType {
+        println!("declared package type");
+        PackageType::Cargo
+    }
+
+    fn is_path_lockfile(&self, path: &Path) -> bool {
+        path.file_name() == Some(OsStr::new("Cargo.lock"))
+    }
+}
+
+impl From<Package> for PackageDescriptor {
+    fn from(package: Package) -> Self {
+        let source = package.source.unwrap_or_default();
+        let version = if source.starts_with("git+") {
+            source.trim_start_matches("git+").into()
```

Comment:
I don't think this is what we want. It would trim `git+git+git+git+` for example?

Looks like what you want is `if let Some(git_version) = source.strip_prefix("git+")`?

Reply:
Fixed.